ui/src/side_nav/components/UserNavBlock.js

```diff
@@ -1,15 +1,18 @@
 import React, {PropTypes, Component} from 'react'
 import {connect} from 'react-redux'
 import {bindActionCreators} from 'redux'
+import {withRouter} from 'react-router'
 
 import classnames from 'classnames'
 
 import {meChangeOrganizationAsync} from 'shared/actions/auth'
 
 class UserNavBlock extends Component {
-  handleChangeCurrentOrganization = organizationID => () => {
-    const {links, meChangeOrganization} = this.props
-    meChangeOrganization(links.me, {organization: organizationID})
+  handleChangeCurrentOrganization = organizationID => async () => {
+    const {router, links, meChangeOrganization} = this.props
+
+    await meChangeOrganization(links.me, {organization: organizationID})
+    router.push('')
```

Comment:
Same here.  This is stateless so lets make it an SFC.

Reply:
It is done